conda_concourse_ci/execute.py

```diff
@@ -1033,33 +1059,9 @@ def _get_activate_builds(concourse_url, limit):
 
 
 def rm_pipeline(pipeline_names, config_root_dir, do_it_dammit=False, pass_throughs=None, **kwargs):
-    config_path = os.path.expanduser(os.path.join(config_root_dir, 'config.yml'))
-    with open(config_path) as src:
-        data = yaml.load(src)
-
-    # make sure we are logged in to the configured server
-    login_args = ['fly', '-t', 'conda-concourse-server', 'login',
-                  '--concourse-url', data['concourse-url'],
-                  '--team-name', data['concourse-team']]
-    if 'concourse-username' in data:
-        # auth is optional.  With Github OAuth, there's an interactive prompt that asks
-        #   the user to go log in with a web browser.  This should not interfere with that.
-        login_args.extend(['--username', data['concourse-username'],
-                           '--password', data['concourse-password']])
-
-    subprocess.check_call(login_args)
+    _ensure_login_and_sync(config_root_dir)
 
-    # sync (possibly update our client version)
-    subprocess.check_call('fly -t conda-concourse-server sync'.split())
-
-    existing_pipelines = subprocess.check_output('fly -t conda-concourse-server ps'.split())
-    if hasattr(existing_pipelines, 'decode'):
-        existing_pipelines = existing_pipelines.decode()
-    existing_pipelines = [line.split()[0] for line in existing_pipelines.splitlines()[1:]]
-
-    pipelines_to_remove = []
-    for pattern in ensure_list(pipeline_names):
-        pipelines_to_remove.extend([p for p in existing_pipelines if fnmatch(p, pattern)])
+    pipelines_to_remove = _filter_existing_pipelines(pipeline_names)
 
     print("Removing pipelines:")
```

Comment:
Should abort_pipeline be called prior to removing the pipeline?  I think it may not be necessary, but I've always wondered how the workers handle pipelines that just disappear.  Do they stop immediately, or do they finish?  If it's the latter, adding in aborting may help things.

Reply:
I think it should be called, because I have seen stray processes consuming resources on the worker nodes, even though the pipelines had been deleted.